pkg/kubelet/rkt/rkt.go

```diff
@@ -1357,6 +1457,30 @@ func (r *Runtime) KillPod(pod *api.Pod, runningPod kubecontainer.Pod, gracePerio
 		return err
 	}
 
+	// `pod` can be nil if it no longer exists in the API and is now being killed
+	var netCleanupPod *api.Pod
+	if pod != nil {
+		if !kubecontainer.IsHostNetworkPod(pod) {
+			netCleanupPod = pod
+		}
+	} else {
+		// Just use the running pod details
+		netCleanupPod = &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				UID:       runningPod.ID,
+				Name:      runningPod.Name,
+				Namespace: runningPod.Namespace,
+			},
+		}
+	}
+
+	if netCleanupPod != nil {
+		if err := r.cleanupPodNetwork(netCleanupPod); err != nil {
+			glog.Errorf("rkt: failed to tear down network for unit %q: %v", serviceName, err)
+			return err
+		}
+	}
+
```

Comment:
Can we move the cleanup code snippets after `if res != "done"` ?
Also seems the `podUID`, `podName`, `podNamespace` are the only required fields, so how about simplify the code as:
```go
netCleanupPod = &api.Pod{
    ObjectMeta: api.ObjectMeta{
        UID:       runningPod.ID,
        Name:      runningPod.Name,
        Namespace: runningPod.Namespace,
    },
}

// TODO: Check if the running pod is host network pod.
if pod == nil || !kubecontainer.IsHostNetworkPod(pod) {
    if err := cleanupPodNetwork(netCleanupPod); err != nil {
        ...
    }
}